tools/cov2json/JSONWriter.hpp

```diff
@@ -0,0 +1,58 @@
+//===----------- JSONWriter.hpp - Serializes structures to JSON -----------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef JSONWriter_hpp
+#define JSONWriter_hpp
+
+#include "swift/Basic/JSONSerialization.h"
+#include "ProfileData.hpp"
+
+namespace swift {
+namespace json {
+
+template <typename U> struct ArrayTraits<std::vector<U>> {
+  static size_t size(Output &out, std::vector<U> &seq) { return seq.size(); }
+  static U &element(Output &out, std::vector<U> &seq, size_t index) {
+    if (seq.size() <= index) {
+      seq.resize(index + 1);
+    }
+    return seq[index];
+  }
+};
+
+template <> struct ObjectTraits<llvm::cov2json::Function> {
+  static void mapping(Output &out, llvm::cov2json::Function &function) {
+    uint64_t regionCount = function.regionCount();
+    uint64_t executedCount = function.executedRegionCount();
+    out.mapRequired("name", function.name);
+    out.mapRequired("symbol", function.symbol);
+    out.mapRequired("regions", regionCount);
+    out.mapRequired("exec_regions", executedCount);
```

Comment:
Can we avoid abbreviations?  (Ambiguities are possible, for example, does `exec_regions` stand for "executable regions"?)